app/services/storage.js

```diff
@@ -29,13 +29,51 @@ export default Service.extend({
     return !!this.getItem('travis.auth.become');
   },
 
+  get billingStep() {
+    return +this.getItem('travis.billing_step');
+  },
+
+  get billingInfo() {
+    try {
+      return JSON.parse(this.getItem('travis.billing_info')) || {};
+    } catch (e) {
+      return {};
+    }
+  },
+
+  get billingPlan() {
+    try {
+      return JSON.parse(this.getItem('travis.billing_plan')) || {};
+    } catch (e) {
+      return {};
+    }
+  },
+
+  set billingInfo(value) {
+    this.setItem('travis.billing_info', JSON.stringify(value));
+  },
+
+  set billingPlan(plan) {
+    this.setItem('travis.billing_plan', JSON.stringify(plan));
+  },
+
+  set billingStep(step) {
+    this.setItem('travis.billing_step', +step);
+  },
+
```

Comment:
Looks awesome 🎉 Almost perfect 👍The only thing that could make sense is to rearrange the code a bit. It's usually convenient to have setters right next to getters, because basically they describe the same property (getters and setters are basically just a syntactic sugar for property descriptors.

 And also, just to make it absolutely perfect - it could make sense to extract parsing to a separate helper method, which will help to make the code more DRY:

```js
get billingInfo() {
 return this.parseWithDefault('travis.billing_info', {});
}

// helpers

parseWithDefault(key, defaultValue) {
  try {
    return JSON.parse(this.getItem(key, defaultValue)) || ;
  } catch (e) {
    return defaultValue;
  }
}
```